### ISSUE TITLE
confusion for experiment on large dataset

### ISSUE THREAD

**Issue by stanpcf** [FlorentAvellaneda/InferDT#1]
Hi, thx for your work. 
but the dataset for your experiment is too small. ml algorithm on large dataset will be confident. 
so I want to know the metric on large dataset such as lightgbm's experiment https://lightgbm.readthedocs.io/en/latest/Experiments.html

**Comment by FlorentAvellaneda** [comment]
Thank you for your message.

The complexity of my algorithm mostly depends on:
-  the size of the model to be inferred (i.e. the minimum tree depth to be consistent with the training data)

- the number of binarized features (i.e. the number of different values that the different features can take)

As my algorithm works iteratively, the number of examples in the learning set has very little impact on performance.
In my experiments, the algorithm seems to be able to infer optimal decision trees with a dozen levels of depth if the number of features is counted in tens, and half this depth if the number of features is counted in thousands.

In the current version of my algorithm, the goal is to find a minimal decision tree with an accuracy of 100% on the learning data set.
So we assume that the data is noiseless and deterministic (there can't be two elements present in two different classes with the same features).
The datasets you propose do not seem to satisfy these criteria, so the minimal tree with 100% accuracy on the dataset should have a depth well over a dozen levels.

The question of finding optimal trees for noisy/non-deterministic datasets is very interesting, and this is the question I'm currently working on.

**Comment by stanpcf** [comment]
thx for your reply and research.
